src/Raku/ast/code.rakumod

```diff
@@ -1821,11 +1822,61 @@ class RakuAST::Sub
         $signature ?? $signature.provides-return-value !! False
     }
 
+    method PERFORM-CHECK(Resolver $resolver, RakuAST::IMPL::QASTContext $context) {
+        return Nil unless self.multiness eq 'multi';
+
+        self.IMPL-CHECK-FOR-DUPLICATE-MULTI-SIGNATURES($resolver);
+    }
+
     method IMPL-COMPILE-BODY(RakuAST::IMPL::QASTContext $context) {
         self.IMPL-WRAP-RETURN-HANDLER($context,
             self.IMPL-WRAP-SCOPE-HANDLER-QAST($context,
                 self.IMPL-APPEND-SIGNATURE-RETURN($context, $!body.IMPL-TO-QAST($context))))
     }
+
+    method IMPL-CHECK-FOR-DUPLICATE-MULTI-SIGNATURES(Resolver $resolver) {
+        my $proto := self.meta-object.dispatcher;
+        my $signature := self.signature.compile-time-value;
+
+        # If we ourselves can default, there is no need to check further
+        return Nil if self.meta-object.can("default");
+
+        my int $has-post-constraints;
+        for self.IMPL-UNWRAP-LIST($signature.params) {
+            $has-post-constraints := 1 if $_.constraint_list;
+        }
+
+        my @seen-accepts;
+        for $proto.dispatchees {
+            last if $_.can("default");
+
+            my $other-signature := $_.signature;
+            next unless $signature.arity == $other-signature.arity;
+
+            my $other-has-post-constraints;
+            for self.IMPL-UNWRAP-LIST($other-signature.params) {
+                $other-has-post-constraints := 1 if $_.constraint_list
+            }
+
+            @seen-accepts.push($_)
+                if !($has-post-constraints || $other-has-post-constraints)
+                        && ($other-signature.ACCEPTS($signature) && $signature.ACCEPTS($other-signature));
+
+            if @seen-accepts > 1 {
```

Comment:
You could `next if $_ =:= $meta-object` early in the loop to avoid comparing this method with itself (which should always match anyway) and then change this to `@seen-accepts > 0`

Reply:
Can't believe I missed this one! Fixed in latest.